File: src/main/java/org/jukeboxmc/block/BlockChorusFlower.java
package org.jukeboxmc.block;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockChorusFlower extends Block {

    public BlockChorusFlower() {
        super( "minecraft:chorus_flower" );
    }

    public void setAge( int value ) {
        this.setState( "age", value );
    }

    public int getAge() {
        return this.stateExists( "age" ) ? this.getIntState( "age" ) : 0;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockChorusFlower.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.ChorusFlower
import org.jukeboxmc.server.block.JukeboxBlock

class BlockChorusFlower(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates),
    ChorusFlower {

    override fun getAge(): Int {
        return this.getIntState("age")
    }

    override fun setAge(value: Int): BlockChorusFlower {
        return this.setState("age", value)
    }
}